### PR TITLE
Add DisableEscapeHTML option for json logger

### PR DESCRIPTION
**Pull request by soh335**
Hi.
I add DisableEscapeHTML option for json logger. If pass ```log.DisableEscapeHTML(true)``` to NewJSONLogger, we can get more readability.

Thanks.

### CODE REVIEW

**Comment by soh335** [comment]
oops. sorry I break ```func(io.Writer) log.Logger``` interface. 

I think there are some ideas.

* add like a ```NewJSONLoggerWithOption``` ( other constructor function dont contain WithOption. it seems to be bad)
* create Option interface and define constructor function as  ```func(w: io.Writer, options ...Options)``` ( need to change other loggers )
* call SetHTMLEscape(false) in jsonLogger without option. ( breaking change )
* other idea ?

**Comment by ChrisHines** [comment]
My initial thought is to call SetHTMLEscape(false) in jsonLogger without option. I'm not sure how much that is really a breaking change. The output will still be JSON compliant. I wouldn't consider the browser as a primary target for the output of jsonLogger.

**Comment by soh335** [comment]
>I wouldn't consider the browser as a primary target for the output of jsonLogger.

I think so too. so no one care about this change, call SetHTMLEscape(false) in jsonLogger seems to be better than others.

**Comment by nicolaiskogheim** [comment]
`enc.SetEscapeHTML(!l.disableEscapeHTML)` (log/json_logger.go:50)

This would read better if the property was named `escapeHTML` instead, I think.

**Comment by soh335** [comment]
@peterbourgon how about this ?

File: log/json_logger.go [hunk 1]
```diff
@@ -10,14 +10,29 @@ import (
 
 type jsonLogger struct {
 	io.Writer
+	disableEscapeHTML bool
```
Comment (peterbourgon) [inline log/json_logger.go]: This should be 

```go
escapeHTML bool
```

defaulting to false.

File: log/json_logger.go [hunk 2]
```diff
@@ -10,14 +10,29 @@ import (
...
+		option(l)
+	}
+	return l
+}
+
+// JSONLoggerOption sets a parameter for json logger
+type JSONLoggerOption func(*jsonLogger)
+
+// DisableEscapeHTML disable to escape &, <, >.
+func DisableEscapeHTML(v bool) JSONLoggerOption {
```
Comment (peterbourgon) [inline log/json_logger.go]: This should be 

```go
func SetEscapeHTML(on bool) JSONLoggerOption {
```

mirroring [json/Encoder.SetEscapeHTML](https://golang.org/pkg/encoding/json/#Encoder.SetEscapeHTML).

File: log/json_logger.go [hunk 3]
```diff
@@ -31,7 +46,9 @@ func (l *jsonLogger) Log(keyvals ...interface{}) error {
 		}
 		merge(m, k, v)
 	}
-	return json.NewEncoder(l.Writer).Encode(m)
+	enc := json.NewEncoder(l.Writer)
+	enc.SetEscapeHTML(!l.disableEscapeHTML)
```
Comment (peterbourgon) [inline log/json_logger.go]: This should be

```go
enc.SetEscapeHTML(l.escapeHTML)
```

**Comment by soh335** [comment]
@peterbourgon thanks for reviewing and change to SetEscapeHTML.
/log/level/level_test.go and log/term/example_test.go is failed because log.NewJSONLogger is not ```func(io.Writer) log.Logger``` interface.

Should I change like this ? (I think its not smart...)

```diff
-                       format: log.NewJSONLogger,
+                       format: func(w io.Writer) log.Logger { return log.NewJSONLogger(w) },
```

```diff
-       logger := term.NewLogger(os.Stdout, log.NewJSONLogger, colorFn)
+       logger := term.NewLogger(os.Stdout, func(w io.Writer) log.Logger { return log.NewJSONLogger(w) }, colorFn)
```

**Comment by soh335** [comment]
Is it mean call ```SetEscapeHTML(false)``` in json logger and call like ```html.EscapeString(v)``` function in HTMLEscapeLogger ?
I want to just call ```enc.SetEscapeHTML``` (its simple), But I know keep interface is also important...

**Comment by soh335** [comment]
@ChrisHines @peterbourgon it seems to be difficult both to keep ```func(io.Writer) log.Logger```  and add logger options currently. And if dont allow to call enc.SetEscapeHTML without option in json logger, maybe this pr cant be merged. so I will close.

**Comment by ChrisHines** [comment]
I haven't seen any objections to just changing the existing JSONLogger to always call `enc.SetEscapeHTML(false)` and leave the exported API untouched. I would be OK with that change.

**Pull request closed without merging**